source/core/Ky.ts

```diff
@@ -145,14 +149,14 @@ export class Ky {
 		};
 
 		const isRetriableMethod = ky._options.retry.methods.includes(ky.request.method.toLowerCase());
-		const result = isRetriableMethod ? ky._retry(fn) : fn();
+		const result = (isRetriableMethod ? ky._retry(fn) : fn()) as ResponsePromise;
```

Comment:
Sidenote: We should also try to reduce use of `as` as it just overrides the type no matter what.

Reply:
You anyway do that below, so I moved it here

Is it alright?

Reply:
I mean didn't mean to apply that you added it. Was just saying in general, we should try to reduce the `as` whenever possible as they're often as bad as `@ts-expect-error`.

Reply:
Okay, I understand that 👍

Reply:
This way it removes one `@ts-expect-error` so that's better than before

I will keep it ✅